source/utilities/cli.ts

```diff
@@ -0,0 +1,130 @@
+    {bold $} {cyan serve} folder_name
+    {bold $} {cyan serve} [-l {underline listen_uri} [-l ...]] [{underline directory}]
+
+    By default, {cyan serve} will listen on {bold 0.0.0.0:3000} and serve the
+    current working directory on that address.
+
+    Specifying a single {bold --listen} argument will overwrite the default, not supplement it.
+
+  {bold OPTIONS}
+
+    --help                              Shows this help message
+
+    -v, --version                       Displays the current version of serve
+
+    -l, --listen {underline listen_uri}             Specify a URI endpoint on which to listen (see below) -
+                                        more than one may be specified to listen in multiple places
+
+    -p                                  Specify custom port
+
+    -d, --debug                         Show debugging information
+
+    -s, --single                        Rewrite all not-found requests to \`index.html\`
+
+    -c, --config                        Specify custom path to \`serve.json\`
+
+    -C, --cors                          Enable CORS, sets \`Access-Control-Allow-Origin\` to \`*\`
+
+    -n, --no-clipboard                  Do not copy the local address to the clipboard
+
+    -u, --no-compression                Do not compress files
+
+    --no-etag                           Send \`Last-Modified\` header instead of \`ETag\`
+
+    -S, --symlinks                      Resolve symlinks instead of showing 404 errors
+    
+    --ssl-cert                          Optional path to an SSL/TLS certificate to serve with HTTPS
+    
+    --ssl-key                           Optional path to the SSL/TLS certificate\'s private key
+
+    --ssl-pass                          Optional path to the SSL/TLS certificate\'s passphrase
+
+    --no-port-switching                 Do not open a port other than the one specified when it\'s taken.
+
+  {bold ENDPOINTS}
+
+    Listen endpoints (specified by the {bold --listen} or {bold -l} options above) instruct {cyan serve}
+    to listen on one or more interfaces/ports, UNIX domain sockets, or Windows named pipes.
+
+    For TCP ports on hostname "localhost":
+
+      {bold $} {cyan serve} -l {underline 1234}
+
+    For TCP (traditional host/port) endpoints:
+
+      {bold $} {cyan serve} -l tcp://{underline hostname}:{underline 1234}
+
+    For UNIX domain socket endpoints:
+
+      {bold $} {cyan serve} -l unix:{underline /path/to/socket.sock}
+
+    For Windows named pipe endpoints:
+
+      {bold $} {cyan serve} -l pipe:\\\\.\\pipe\\{underline PipeName}
+`;
+
+/**
+ * Parses the program's `process.argv` and returns the options and arguments.
+ *
+ * @returns The parsed options and arguments.
+ */
+export const parseArguments = (): Arguments =>
+  // @ts-expect-error The handler array for the `--listen` option does have only
+  // one element, I'm not sure why Typescript is throwing an error here.
+  parseArgv(options) as unknown as Arguments;
```

Comment:
It's because TS knows that an array is not always guaranteed.

You could remove the array or force the type:
```ts
  '--listen': parseEndpoint,
  '--listen': [parseEndpoint] as [typeof parseEndpoint],
```

This then surfaces another issue, which would be great to fix.
```
Argument of type '{ '--help': BooleanConstructor; '--version': BooleanConstructor; '--listen': (uriOrPort: ListenEndpoint) => ParsedEndpoint; ... 22 more ...; '-p': string; }' is not assignable to parameter of type 'Spec'.
  Property ''--listen'' is incompatible with index signature.
    Type '(uriOrPort: ListenEndpoint) => ParsedEndpoint' is not assignable to type 'string | Handler<any> | [Handler<any>]'.
      Type '(uriOrPort: ListenEndpoint) => ParsedEndpoint' is not assignable to type 'Handler<any>'.
        Types of parameters 'uriOrPort' and 'value' are incompatible.
```

Reply:
The function signature of `parseEndpoint` wasn't compatible with the `Handler` signature, so I fixed that.